### PR TITLE
tests plus fix for mangling order of grouped items

### PR DESCRIPTION
**Pull request by rsheeter**

### CODE REVIEW

File: src/picosvg/svg.py [hunk 1]
```diff
@@ -280,13 +280,13 @@ def _resolve_clip_path(self, clip_path_url):
...
         # multiple clip paths leave behind their intersection
-        if len(clip_paths) > 1:
-            return svg_pathops.intersection(*clip_paths)
-        elif clip_paths:
-            return clip_paths[0]
-        return None
+        if not clip_paths:
+            raise ValueError("Cannot combine no clip_paths")
+        if len(clip_paths) == 1:
+            clip_paths[0]
```
Comment (anthrotype) [inline src/picosvg/svg.py:288]: what's this statement doing? was it meant to return?

Comment (rsheeter) [reply src/picosvg/svg.py:288]: ...yes!

File: src/picosvg/svg.py [hunk 2]
```diff
@@ -354,7 +354,10 @@ def _ungroup(self, scope_el):
         multi_clips = []
         for group in groups:
             # move groups children up
-            for child in group:
+            # reverse because "for each addnext" effectively reverses
+            children = [c for c in group]
```
Comment (anthrotype) [inline src/picosvg/svg.py:358]:
```suggestion
            children = list(group)
```

File: src/picosvg/svg.py [hunk 3]
```diff
@@ -724,7 +729,8 @@ def fromstring(cls, string):
             string = string.replace("xlink:href", _XLINK_TEMP)
 
         # encode because fromstring dislikes xml encoding decl if input is str
-        tree = etree.fromstring(string.encode('utf-8'))
+        parser = etree.XMLParser(remove_blank_text=True)
```
Comment (anthrotype) [inline src/picosvg/svg.py:732]: nice. will this remove the need for the extra stripping in picosvg module?

**Pull request merged**